app/src/common/hooks/useEditMapping.tsx

```diff
@@ -0,0 +1,115 @@
+import { useCallback, useEffect, useState } from "react";
+
+import { useParams } from "react-router-dom";
+
+import { useAppDispatch } from "app/store";
+import { columnsAdded, columnsRemoved } from "features/Columns/columnSlice";
+import { filtersAdded, filtersRemoved } from "features/Filters/filterSlice";
+import { joinsAdded, joinsRemoved } from "features/Joins/joinSlice";
+import {
+  resourceAdded,
+  resourcesRemoved,
+} from "features/Mappings/resourceSlice";
+import {
+  useApiColumnsListQuery,
+  useApiFiltersListQuery,
+  useApiJoinsListQuery,
+  useApiResourcesRetrieveQuery,
+} from "services/api/endpoints";
+import {
+  Column,
+  Filter,
+  Join,
+  Resource,
+} from "services/api/generated/api.generated";
+
+/**
+ * Initialize the local store with resource, filters, joins & columns related to the URL search param "mappingId"
+ *
+ * Resets the store on unmount (resourceSlice, filterSlice, joinSlice, columnSlice)
+ */
+const useEditMapping = (): {
+  isLoading: boolean;
+  data?: {
+    resource: Resource;
+    filters: Filter[];
+    joins: Join[];
+    columns: Column[];
+  };
+} => {
+  const dispatch = useAppDispatch();
+  const { mappingId } = useParams<{ mappingId?: string }>();
+  const { data: mapping } = useApiResourcesRetrieveQuery(
+    { id: mappingId ?? "" },
+    { skip: !mappingId }
+  );
+  const { data: mappingFilters } = useApiFiltersListQuery(
+    { resource: mappingId },
+    { skip: !mappingId }
+  );
+  const { data: joins } = useApiJoinsListQuery({}, { skip: !mappingId });
+  const { data: columns } = useApiColumnsListQuery({}, { skip: !mappingId });
+
+  const [isMappingInit, setMappingInit] = useState(false);
+  const [isFiltersInit, setFiltersInit] = useState(false);
+  const [isJoinsInit, setJoinsInit] = useState(false);
+  const [isColumnsInit, setColumnsInit] = useState(false);
```

Comment:
you could use selectors here instead of useState no?

Reply:
Maybe, to check if requested items are in store? 

I need to make sure dispatches are made only once per item type

Reply:
I am not sure I can rely on store values.. They can totally change during this hook lifetime

Reply:
I've tried this instead : 

```
const {
    isMappingInit,
    isFiltersInit,
    isJoinsInit,
    isColumnsInit,
  } = useAppSelector((state) => ({
    isMappingInit:
      mapping &&
      isEqual(resourceSelectors.selectById(state, mapping.id), mapping),
    isFiltersInit:
      mappingFilters &&
      isEqual(filterSelectors.selectAll(state), mappingFilters),
    isJoinsInit: joins && isEqual(joinSelectors.selectAll(state), joins),
    isColumnsInit:
      columns && isEqual(columnSelectors.selectAll(state), columns),
  }));
```
But as soon as values are changed by user, this doesn't work

I also can't check on whether slices are empty or not, because there can simply be none joins/filters/columns in the current mapping

Reply:
I don't get why you use isEqual here. 
```typescript
{
  isMappingInit: mapping && !!resourceSelectors.selectById(state, mapping.id), // it returns undefined if not yet init
  isFiltersInit: filterSelectors.selectAll(state).length > 0, // should return an empty array if not init
  ...
}
```

Reply:
Sure, mapping will still be in the store.

But `filterSelectors.selectAll(state)` can be empty even though everything is dispatched. The same can occur for joins and  columns